### ISSUE TITLE
Sokol + Imgui + emscripten `InputText` not working.

### ISSUE THREAD

**Issue by voidware** [floooh/sokol#809]
Cannot enter any text in an Imgui `InputText` widget on mobile using Firefox browser (only one tested).

Keyboard appears. Text appears, but _Enter_ does not work. so the line cannot be submitted.

**Comment by voidware** [comment]
Ok, false alarm. Or rather i forgot to patch in my previous hacks when updating Sokol. This should really be sorted.

Here's the fix:

Use `event->key` not `event->code`. I know apparently code supersedes key, but it doesn't work!

`sokol_app.h` 

```
_SOKOL_PRIVATE EM_BOOL _sapp_emsc_key_cb(int emsc_type, const EmscriptenKeyboardEvent* emsc_event, void* user_data) {
...
            else {
                
                // XXX HACK VOIDWARE
                //_sapp.event.key_code = _sapp_emsc_translate_key(emsc_event->code);
                _sapp.event.key_code = _sapp_emsc_translate_key(emsc_event->key);
                ...

```

**Comment by floooh** [comment]
Chrome on Android seems to have the same behaviour.

Did you write your own version of ```_sapp_emsc_translate_key()```? Because the function expects an integer key code, not a string (```emsc_event->key``` is a string).

**Comment by floooh** [comment]
...ah ok, the TextInput field **does** take the char code from SAPP_EVENTTYPE_CHAR, but only for regular characters, not for special keys like Enter.

The underlying problem remains though that mobile browsers don't send a physical key code (I guess because mobile devices don't have a "physical keyboard", and the only alternative are keys that are already localized, which doesn't allow to map back to the physical key codes).

The only solution would be to treat some keys special as fallback if the key code isn't set (e.g. only send a handful of special keys which are not affected by localization, like Enter, Arrows, etc...).

**Comment by floooh** [comment]
(PS: on Android Chrome I actually don't get any input in the Dear ImGui text input field... I'll try Android Firefox next)

**Comment by floooh** [comment]
PS: reading up on html5 keyboard input here: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent...

...it looks like all event-based methods to get UNICODE character input are deprecated, and the only intended way is to use a HTML5 text input field and inspect its contents after a key press event is received.

sokol_app.h opens such a hidden text input field, but this comes with all sorts of its own problems (e.g. IIRC on Safari this isn't allowed to be hidden for fishing-protection purposes).

Long story short: I would not recommend to depend too much on working text input on mobile devices, because modern HTML5 will always find a away to sabotage getting international text input without all sorts of hair raising hacks.

**Comment by floooh** [comment]
=> see https://github.com/floooh/sokol/issues/812
